Split into a multi page site using Vitepress

OceanIsEndless opened the pull request:
Fixes #2, and is a continuation of pull request #3. Vitepress has now been set up, but only a couple major sections have been split so far. Much more is to come! Thanks to @gradylink for helping along the process.

OceanIsEndless committed:
Split into a multi page site using Vitepress

Fixes #2, and is a continuation of pull request #3. Vitepress has now been set up, but only a couple major sections have been split so far. Much more is to come.

OceanIsEndless committed:
fix some links

OceanIsEndless committed:
Set up Pages for Vitepress

OceanIsEndless committed:
Here we go again

OceanIsEndless committed:
testing testing 123

gradylink commented:
it's kinda funny 'cause you'll keep trying things until you see this...

OceanIsEndless committed:
Fix the action

(maybe)

OceanIsEndless commented:
Hmm, seems like it needs to install vitepress or something?? Sorry I'm dumb lol

gradylink commented:
You have `npm`/`node` installed right?

OceanIsEndless committed:
Fix the action for real

(probably)

OceanIsEndless commented:
Progress was made! Now it's complaining about my images :>

gradylink commented:
> It added node_modules, which I guess must be an essential component?... and it added a bunch of stuff to the package-lock.json! I have a good feeling about this now. (Seemed like something was missing there.)

just don't commit and push `node_modules`.

OceanIsEndless commented:
> just don't commit and push `node_modules`.

I've already added it to the .gitignore! Was able to tell it shouldn't be there from your original pull.

> I've never seen that error before...

Uh-oh

Maybe it's something to do with how the images were put in the markdown? (idk, the error seems to be referencing them)

```md
#### Rotation Style: All Around

![A demonstration of the "all around" rotation style](/img/all-around.gif)

#### Rotation Style: Left-Right

![A demonstration of the "left-right" rotation style](/img/left-right.gif)

#### Rotation Style: Don't Rotate

![A demonstration of the "don't rotate" rotation style](/img/don't-rotate.gif)
```

OceanIsEndless commented:
> yeah, i'll see if I can figure it out. Also I think I might make a custom markdown extension to handle scratchblocks code blocks.

Epic!

> you should delete the 2 markdown files in `.vitepress/`

Apologies, I have no idea why I thought moving the example files to the config dir wouldn't cause any issues 🥲

> replace `{{toc}}` with `[[toc]]`

Shoot, I thought I caught that.

OceanIsEndless committed:
wahoo

OceanIsEndless commented:
I went back to the docs, maybe this is the problem?
https://vitepress.dev/guide/asset-handling#the-public-directory

> Common image, media, and font filetypes are detected and included as assets automatically.
>
> > PDFs or other documents referenced by links within markdown files are not automatically treated as assets. To make linked files accessible, you must manually place them within the [public](https://vitepress.dev/guide/asset-handling#the-public-directory) directory of your project.

Maybe it's not recognizing the .gifs or something dumb like that? Just throwing out ideas. I might try omitting the images for now and see if it works.

OceanIsEndless committed:
try deploying w/o images

OceanIsEndless commented:
It doesn't like my broken heading links either 8>

It says I can disable the check for them though... gonna try it

OceanIsEndless committed:
ignore dead links

OceanIsEndless commented:
This is why I don't like deployments lol

Seems to be just a permissions issue now though. The build worked! Might have misconfigured something in the repository that is stopping the deployment. Just trying to get it deployed, and then it can be refined later.

OceanIsEndless commented:
Fixed settings issue with deploying (I think; [random answer online](https://github.com/orgs/community/discussions/39054#discussioncomment-6420042)), going to manually run and see if it will deploy!

OceanIsEndless commented:
...I lack the mental capacity to run it manually, triggering by push 😐

OceanIsEndless committed:
deploy sesame!

OceanIsEndless commented:
🎉

The image thing is still unresolved, but it worked!! Yay!!!

gradylink commented:
OK, i got the markdown-it plugin to work but all the blocks are black for some reason?

gradylink commented:
<img width="658" height="506" alt="image" src="https://github.com/user-attachments/assets/3a18a5d9-732c-4097-ab27-06cc0be1fcf7" />

OK I got it working. What you need to do is run `npm install sb-mdit` and then replace your `.vitepress/config.mts` with:

```typescript
import { defineConfig } from 'vitepress'
import { scratchblocksPlugin } from 'sb-mdit'

// https://vitepress.dev/reference/site-config
export default defineConfig({
  title: "Scratch Specification",
  base: "/scratch-spec/", // https://vitepress.dev/guide/deploy#setting-a-public-base-path
  ignoreDeadLinks: true,

  themeConfig: {
    // https://vitepress.dev/reference/default-theme-config
    nav: [
      { text: 'Home', link: '/' },
      { text: 'Intro', link: '/intro/' },
      { text: 'Concepts', link: '/concepts/' },
      { text: 'Runtime', link: '/runtime/' },
      { text: 'Blocks', link: '/palette/' },
      { text: 'Files', link: '/files/' },
      { text: 'I/O', link: '/io/' },
      { text: 'Network', link: '/network/' },
      { text: 'Devices', link: '/devices/' }
    ],

    sidebar: [
      {
        text: 'Introduction',
        link: '/intro/',
        items: [
          { text: 'Frequently Asked Questions', link: '/intro/#faq' },
          { text: 'Contributing Guidelines', link: '/intro/#contributing' }
        ]
      },
      {
        text: 'Concepts',
        link: '/concepts/',
        items: [
          { text: 'Capabilities', link: '/concepts/#capabilities' },
          { text: 'Constants', link: '/concepts/#constants' },
          { text: 'Ideas', link: '/concepts/#ideas' },
          { text: 'Values', link: '/concepts/#values' },
          { text: 'Procedures', link: '/concepts/#procedures' }
        ]
      },
    ],

    socialLinks: [
      { icon: 'github', link: 'https://github.com/OceanIsEndless/scratch-spec' }
    ],

    search: {
      provider: 'local'
    }
  },
  markdown: {
    config: md => md.use(scratchblocksPlugin)
  }
})
```

OceanIsEndless committed:
add plugins

Awesome! Thanks so much for this:
https://github.com/gradylink/sb-mdit

gradylink commented:
Oh, another suggestion. Move contents onto its own page. It's kind weird on the homepage

OceanIsEndless committed:
fix assets not importing

After some tinkering, I finally realized that the apostrophe in the name "don't-rotate.gif" was causing the error with vitepress! It put a single quote inside a single-quoted string...

// code in the internal error
import _imports_2 from '/img/don't-rotate.gif'

vitepress complained that it was invalid JS syntax because, well, it was! Yes, this is probably an error that shouldn't happen in the first place, but it's amazing how little things like that can mess it all up.

For now, the apostrophe has been removed from the file name, and it now properly loads. Additionally, all assets have been moved to the public folder for easier organization.

OceanIsEndless committed:
break off Contents page

per @gradylink's suggestion

OceanIsEndless committed:
Quickly fix Contents link

gradylink commented:
You should submit a bug to vitepress about the image issue.

OceanIsEndless committed:
add dark mode friendly rotation style graphics

And also just nicer graphics in general

OceanIsEndless committed:
separate blocks page and adjust stuff

OceanIsEndless committed:
finish move from README

OceanIsEndless committed:
lots of stuff

A LOOOOOT of stuff

I do not wish to elaborate

OceanIsEndless committed:
the tiniest change ever

a welcome relief after all the other changes added in the previous commit

OceanIsEndless commented:
oh shoot

OceanIsEndless commented:
there :D
